### ISSUE TITLE
Build failed with -DCMAKE_BUILD_TYPE=Debug under Ubuntu14

### ISSUE THREAD

**Issue by flm8620** [nmoehrle/mvs-texturing#58]
I have build this project using cmake under Ubuntu14, no problem.

But when I add -DCMAKE_BUILD_TYPE=Debug, I got errors.

I launched the build from Qt Creator. Then I tried to build it from terminal, but it gives the same error.

```
11:13:40: Running steps for project Texturing...
11:13:40: Starting: "/usr/bin/cmake" --build . --target all
[ 14%] Built target ext_eigen
[ 29%] Built target ext_mve
[ 44%] Built target ext_rayint
[ 51%] Built target mrf
Scanning dependencies of target tex
[ 53%] Building CXX object libs/tex/CMakeFiles/tex.dir/generate_texture_views.cpp.o
[ 55%] Building CXX object libs/tex/CMakeFiles/tex.dir/timer.cpp.o
[ 57%] Building CXX object libs/tex/CMakeFiles/tex.dir/global_seam_leveling.cpp.o
[ 59%] Building CXX object libs/tex/CMakeFiles/tex.dir/texture_view.cpp.o
[ 61%] Building CXX object libs/tex/CMakeFiles/tex.dir/build_obj_model.cpp.o
[ 62%] Building CXX object libs/tex/CMakeFiles/tex.dir/tri.cpp.o
[ 64%] Building CXX object libs/tex/CMakeFiles/tex.dir/poisson_blending.cpp.o
[ 66%] Building CXX object libs/tex/CMakeFiles/tex.dir/generate_texture_patches.cpp.o
[ 68%] Building CXX object libs/tex/CMakeFiles/tex.dir/material_lib.cpp.o
[ 70%] Building CXX object libs/tex/CMakeFiles/tex.dir/calculate_data_costs.cpp.o
[ 72%] Building CXX object libs/tex/CMakeFiles/tex.dir/seam_leveling.cpp.o
[ 74%] Building CXX object libs/tex/CMakeFiles/tex.dir/texture_atlas.cpp.o
[ 75%] Building CXX object libs/tex/CMakeFiles/tex.dir/histogram.cpp.o
[ 77%] Building CXX object libs/tex/CMakeFiles/tex.dir/build_adjacency_graph.cpp.o
[ 79%] Building CXX object libs/tex/CMakeFiles/tex.dir/local_seam_leveling.cpp.o
[ 81%] Building CXX object libs/tex/CMakeFiles/tex.dir/texture_patch.cpp.o
[ 83%] Building CXX object libs/tex/CMakeFiles/tex.dir/prepare_mesh.cpp.o
[ 85%] Building CXX object libs/tex/CMakeFiles/tex.dir/view_selection.cpp.o
[ 87%] Building CXX object libs/tex/CMakeFiles/tex.dir/generate_texture_atlases.cpp.o
[ 88%] Building CXX object libs/tex/CMakeFiles/tex.dir/obj_model.cpp.o
[ 90%] Building CXX object libs/tex/CMakeFiles/tex.dir/generate_debug_embeddings.cpp.o
Linking CXX static library libtex.a
[ 96%] Built target tex
Scanning dependencies of target texrecon
[ 98%] Building CXX object apps/texrecon/CMakeFiles/texrecon.dir/arguments.cpp.o
[100%] Building CXX object apps/texrecon/CMakeFiles/texrecon.dir/texrecon.cpp.o
Linking CXX executable texrecon
../../libs/tex/libtex.a(calculate_data_costs.cpp.o): In function `acc::BVHTree<unsigned int, math::Vector<float, 3> >::bsplit(unsigned int, std::vector<acc::AABB<math::Vector<float, 3> >, std::allocator<acc::AABB<math::Vector<float, 3> > > > const&)':
/home/leman/Documents/mvs-texturing/elibs/rayint/libs/acc/bvh_tree.h:205: undefined reference to `acc::BVHTree<unsigned int, math::Vector<float, 3> >::NAI'
/home/leman/Documents/mvs-texturing/elibs/rayint/libs/acc/bvh_tree.h:205: undefined reference to `acc::BVHTree<unsigned int, math::Vector<float, 3> >::NAI'
../../libs/tex/libtex.a(calculate_data_costs.cpp.o): In function `acc::BVHTree<unsigned int, math::Vector<float, 3> >::ssplit(unsigned int, std::vector<acc::AABB<math::Vector<float, 3> >, std::allocator<acc::AABB<math::Vector<float, 3> > > > const&)':
/home/leman/Documents/mvs-texturing/elibs/rayint/libs/acc/bvh_tree.h:294: undefined reference to `acc::BVHTree<unsigned int, math::Vector<float, 3> >::NAI'
/home/leman/Documents/mvs-texturing/elibs/rayint/libs/acc/bvh_tree.h:294: undefined reference to `acc::BVHTree<unsigned int, math::Vector<float, 3> >::NAI'
collect2: error: ld returned 1 exit status
make[2]: *** [apps/texrecon/texrecon] Error 1
make[1]: *** [apps/texrecon/CMakeFiles/texrecon.dir/all] Error 2
make: *** [all] Error 2
11:14:22: The process "/usr/bin/cmake" exited with code 2.
Error while building/deploying project Texturing (kit: Desktop Qt 5.7.0 GCC 64bit)
When executing step "Make"
11:14:22: Elapsed time: 00:42.
```

**Comment by nmoehrle** [comment]
Yes sorry that is a known problem give me an hour.

**Comment by nmoehrle** [comment]
Sorry I ran into further issues when compiling under Ubuntu 16.04 - working on it...

**Comment by flm8620** [comment]
It's the problem of external library rayint
in rayint/libs/acc/bvh_tree.h
put the definition of NAI outside the class like this:

```
template <typename IdxType, typename Vec3fType>
const  IdxType BVHTree<IdxType, Vec3fType>::NAI = std::numeric_limits<IdxType>::max();
```

and replace the definition in class by:
`const static IdxType NAI;`

**Comment by nmoehrle** [comment]
Yes you are right it took me quite a while to figure it out but I fixed it similar to this in nmoehrle/libacc@d2595c8d. [libacc](https://github.com/nmoehrle/libacc) is the successor of rayint but I havn't taken the time to replace it in mvs-texturing and since yesterday my os is skewed.

Thank you!
